cpu/cc2538/dev/sys-ctrl.h

```diff
@@ -242,6 +242,36 @@
 #endif
 /** @} */
 /*---------------------------------------------------------------------------*/
+/** \name System clock divisor selection
+ * @{
+ */
+#ifdef SYS_CTRL_CONF_SYS_DIV
+#if SYS_CTRL_CONF_SYS_DIV & ~SYS_CTRL_CLOCK_CTRL_SYS_DIV
+#error Invalid system clock divisor
+#else
+#define SYS_CTRL_SYS_DIV        SYS_CTRL_CONF_SYS_DIV
+#endif
+#else
+#define SYS_CTRL_SYS_DIV        SYS_CTRL_CLOCK_CTRL_SYS_DIV_16MHZ
+#endif
+
+#ifdef SYS_CTRL_CONF_IO_DIV
+#if SYS_CTRL_CONF_IO_DIV & ~SYS_CTRL_CLOCK_CTRL_IO_DIV
+#error Invalid IO clock divisor
+#else
+#define SYS_CTRL_IO_DIV         SYS_CTRL_CONF_IO_DIV
+#endif
+#else
+#define SYS_CTRL_IO_DIV         SYS_CTRL_CLOCK_CTRL_IO_DIV_16MHZ
+#endif
+
+/* Returns actual sys clock in MHz */
+#define SYS_CTRL_SYS_CLOCK        (SYS_CTRL_32MHZ >> SYS_CTRL_SYS_DIV)
+/* Returns actual io clock in MHz */
+#define SYS_CTRL_IO_CLOCK       (SYS_CTRL_32MHZ >> (SYS_CTRL_IO_DIV >> 8))
```

Comment:
According to §7.3.1 in the UG, the I/O clock must not exceed 5/3 of the system clock. There should be a `#error` if this is not the case.